fancontrold: Add handling for faulty fan

nunojsa opened the pull request:
Poweroff the device in the case a faulty fan is detected. The shutdown
path is not completely powering off all devices. Hence, before that
the daemon disables all known devices which would be left powered...

Signed-off-by: Nuno Sá <nuno.sa@analog.com>

commodo commented on fancontrold.c:
```diff
@@ -80,12 +86,43 @@ struct monitor_iio_dev_lookup {
 	{"adrv9009-phy-b", 0, "input"},
 };
 
+
```
nitpick: extra line

commodo commented on fancontrold.c:
```diff
@@ -207,6 +244,80 @@ static uint8_t monitor_get_pwm(const long temperature)
+	}
+
+	/* if we reach this point the fan is faulty, let's poweroff! */
+	emerg("FAN is faulty. System is going to poweroff!!!\n");
```
maybe move this print before the `for (dev = 0; dev < ARRAY_SIZE(monitor_shutdown_list); dev++) ` loop
in case there are issues / hangs with one of the shutdowns we get this message

nunojsa replied:
agreed

commodo commented on fancontrold.c:
```diff
@@ -229,6 +340,8 @@ static void run(const time_t sleep)
 		if (!running)
 			break;
 
+		monitor_handle_fan_fault();
```
maybe break the loop here on fault, or do an exit() in `monitor_handle_fan_fault()`
the loop can continue and try stuff [for a short while] as if no-fault

commodo replied:
well, tbh, i found no mention that `reboot()` exits the program

nunojsa replied:
Since we call the `reboot()` syscall i did not care much about that. But it does make sense to call `exit()`.

commodo commented:
rest looks good

nunojsa committed:
fancontrold: Add handling for faulty fan

Poweroff the device in the case a faulty fan is detected. The shutdown
path is not completely powering off all devices. Hence, before that
the daemon disables all known devices which would be left powered...

Signed-off-by: Nuno Sá <nuno.sa@analog.com>

nunojsa commented:
Changes in v2:

- Call `clean_up()` and`exit()` on `monitor_handle_fan_fault()` on a faulty fan;
- Move the `emerg()` log.
